tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -1171,6 +1202,13 @@ void test_main(void)
 			 ztest_unit_test(test_stack_buffer),
 			 ztest_user_unit_test(test_unimplemented_syscall),
 			 ztest_user_unit_test(test_bad_syscall),
+#if !defined(CONFIG_ARC) /* FIXME: #17590 */
```

Comment:
I don't think you can do this because `ztest_test_suite()` itself is a defined macro.  You get: 
```
.../tests/kernel/mem_protect/userspace/./src/main.c: found invalid #endif in ztest_test_suite()
```
 when you run `./scripts/sanitycheck -vvv -p qemu_cortex_m3 -T tests/kernel/mem_protect/userspace`

Reply:
right @yashi , this was overlooked, and spotted, later, also by @andrewboie .- I'll try to work on a quick fix

Reply:
I don't know why this check exists, to be honest